Source/core/JSON.h

```diff
@@ -2809,9 +2811,13 @@ namespace Core {
                         offset = PARSE;
                     }
                 }
-                if ((offset == static_cast<uint16_t>(~0)) && (loaded < maxLength)) {
-                    stream[loaded++] = ']';
-                    offset = 0;
+                if (offset == static_cast<uint16_t>(~0)) {
+                    if (loaded < maxLength) {
+                        stream[loaded++] = ']';
+                        offset = 0;
+                    } else {
+                        offset = END_MARKER;
```

Comment:
This seems a bit scary. END_MARKER is just an enum value of 6. This means that you set the offset here to position 6. If for some reason we would continue to write characters at offset 6, we assume it is the end. Looks like the issue is more on the callee. DO you still know from where this was called ?

Reply:
Sorry, I missed your comments. I will check this